tests/python/relay/test_ir_parser.py

```diff
@@ -0,0 +1,530 @@
+from tvm.relay.parser import enabled
+if not enabled():
+    exit()
+
+import tvm
+from tvm import relay
+from tvm.relay._parser import ParseError
+from tvm.relay.ir_pass import alpha_equal
+from nose.tools import nottest, raises
+from typing import Union
+
+BINARY_OPS = {
+    "*": relay.multiply,
+    "/": relay.divide,
+    "+": relay.add,
+    "-": relay.subtract,
+    "<": relay.less,
+    ">": relay.greater,
+    "<=": relay.less_equal,
+    ">=": relay.greater_equal,
+    "==": relay.equal,
+    "!=": relay.not_equal,
+}
+
+TYPES = {
+    "int8",
+    "int16",
+    "int32",
+    "int64",
+
+    "uint8",
+    "uint16",
+    "uint32",
+    "uint64",
+
+    "float16",
+    "float32",
+    "float64",
+
+    "bool",
+
+    "int8x4",
+    "uint1x4",
+    "float16x4",
+}
+
+def get_scalar(x):
+    # type: (relay.Constant) -> (Union[float, int, bool])
+    return x.data.asnumpy().item()
+
+def is_close(x, y, precision=0.001):
+    return x - y < precision and y - x < precision
+
+int32 = relay.scalar_type("int32")
```

Comment:
try not to construct things in the global env, instead, put them in an env and or construct them in a function

Reply:
I'm not sure what I should do instead. These are used as a shorthands in several of the test functions and so need to be in global scope.